src/style/parse.js

```diff
@@ -374,6 +374,19 @@ styfn.parseImpl = function( name, value, propIsBypass, propIsFlat ){
       bypass: propIsBypass
     };
 
+  } else if( type.colorOrGradient ) {
+    let tuple = util.colorOrGradient2tuple( value );
+    
+    if ( !tuple ){ return null; }
+
+    return {
+      name: name,
+      value: tuple,
+      pfValue: tuple,
+      strValue: '' + value,
+      bypass: propIsBypass
+    };
+    
```

Comment:
Also (4):

(4) The method used for checking whether you have a colour or a gradient will be bad for maintenance.  It would be better to use a flag in the style property, and it seems that the style property parsing could be handled a bit cleaner.  For example, there should be no `colorOrGradient2tuple`.  Colours and gradients should be parsed separately so the style property can better represent which one is detected.